tests/unittest/_torch/attention/sparse/test_sparse_attention.py

```
@@ -0,0 +1,1022 @@
+        **kwargs,
+    ):
+        kwargs["sparse_attention_config"] = MockSparseAttentionConfig()
+        kwargs["pos_embd_params"] = None
+        super().__init__(*args, **kwargs)
+        self._sparse_kv_indices = sparse_kv_indices
+        self._sparse_kv_offsets = sparse_kv_offsets
+        self._sparse_attn_indices = sparse_attn_indices
+        self._sparse_attn_offsets = sparse_attn_offsets
+        self._sparse_attn_ctx_indices = sparse_attn_ctx_indices
+
+    def sparse_kv_predict(self, q, k, metadata, **kwargs):
```

Ruff (0.14.13)

Unused method argument: `q`

(ARG002)